openlibrary/plugins/openlibrary/js/carousels_partials.js

```diff
@@ -29,5 +29,13 @@ export function initCarouselsPartials() {
                 }
             }
         });
-    });
+    };
+
+    $('.loadingIndicator').removeClass('hidden');
+
+    if (document.readyState === 'complete') {
```

Comment:
This should always be true at this point, from my understanding of the code.

Reply:
I was able to enter the `else` block by navigating to a book page using the browser's back button.

Reply:
Sorry meant falsey

Complete: The document and all sub-resources have finished loading. The state indicates that the load event is about to fire.

Else would mean the document (styles, scripts)is still loading or interactive.

I think the only case where this would be true on second thoughts is when this gets lazy loaded last.

Either way I don't think it should matter 
https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState

I do vaguely remember some issues with back button/cached pages and window.onload and older jquery versions so that's possibly what we are hitting here.

Drop the window.load event. Should not be needed